### ISSUE TITLE
Typedef of object with generic property fails since 5.1

### ISSUE THREAD

**Issue by MoonE** [microsoft/TypeScript#54544]
# Bug Report

### 🔎 Search Terms

generic typedef object

### 🕗 Version & Regression Information

- This changed between versions 5.0.4 and 5.1.3

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=5.1.3&filetype=js#code/PQKhAIChwgBAXAngBwKYBNUDNwG8DyARgFaoDG8AvuPtHMgE4D2aDSeAggwwIaIA8AFQB81HnXAJUAW2QAbHvFThBdYJCA)

### 💻 Code

```js
/** 
 * @typedef {Object} O
 * @property {Array<T>} a
 * @template T
 */
```

### 🙁 Actual behavior

Templated properties produce an error since upgrading TS to 5.1.3

### 🙂 Expected behavior

Generic properties should work as in 5.0.4

This still works, but it should be functionally equivalent to the typedef above.
```js
/** 
 * @typedef {{a: Array<T>}} O
 * @template T
 */
```

**Comment by MoonE** [comment]
So it seems it is only a problem when the `@template` is not first. This works fine:

```js
/** 
 * @template T
 * @typedef {Object} O
 * @property {Array<T>} a Array of T
 */
```

**Comment by sandersn** [comment]
The parser just skips it completely. Here's a view from ts-ast-viewer: https://ts-ast-viewer.com/#code/PQKhAIChwgBAXAngBwKYBNUDNwG8DyARgFaoDG8AvuPtHMgE4D2aDSeAggwwIaIA8AFQB81HuC69E4JjkF1wCVAFtkAGx7xU4eTGBA

**Comment by sandersn** [comment]
Yes. In addition, `@template` should end `@typedef` parsing and start JSDocTemplateTag parsing. That required more time for thought than I had when I looked at this.
